### PR TITLE
Improve documentation in config.h (NFC)

### PR DESCRIPTION
**Pull request by sam-mccall**
Adjusted comments that were missing, poorly worded, overly obscure, or
otherwise unhelpful to users. Some research was required!

Removed macros that were unused/trivially used.

Switched /* C comments */ to // line-comments.

Resisted the urge to reorder, rename, and remove options, to keep the
diff small.

### CODE REVIEW

File: config.h [hunk 1]
```diff
@@ -1,216 +1,172 @@
...
+// Some criteria match against the current mode:
+//   `appkey` is linked to whether the keypad is in "application mode"
+//   `appcursor` is linked to whether cursor keys are in "application mode"
+//   `crlf` is linked to whether the terminal is in "automatic newline mode"
+// These have three values
+//   -1: only match if mode is NOT active
+//   0 : match independent of mode
+//   1 : only match if mode is active
+// appkey has one extra value:
+//   2 : only match if mode is active AND numlock was toggled off.
```
Comment (chandlerc) [inline config.h:169]: Isn't 2 triggered when numlock is *on*? Maybe I misread that code.

**Review by chandlerc — CHANGES_REQUESTED** [review]
Awesome, one question below.

File: config.h (hunk 1 above)
Comment (sam-mccall) [reply config.h:169]: That's what the old comment said, but in the `kmap()` loop:
```
if (term.numlock && kp->appkey == 2)
      continue;
```
So if `numlock` is 1, then we *skip* rules with `appkey==2`

Maybe `numlock=1` means *off* and 0 means *on*?

Looking at where this was added in 172f65436ce14a52842d67b862fdc45f8ff3ada3, my interpretation is:
  - this changed the default from generating dedicated escape sequences for `XK_KP_2` and friends, to just generating `2`
  - it seems reasonable to call the new mode numlock, and indeed it was indicated by `numlock=1`
  - so the old comment is wrong and the new one is correct
  - but I have very little confidence the behavior is actually good, worth checking what other terms, apps, tmux think about these keys

Comment (chandlerc) [reply config.h:169]: Wow. Agree that this is an improvement in accuracy, and that we should figure out what is "right" here...

**Review by chandlerc — APPROVED** [review]
Finally back to this, LGTM, merging. Thanks!

**Pull request merged**